### ISSUE TITLE
Value based status monitoring

### ISSUE THREAD

**Issue by moay** [louislam/uptime-kuma#771]
**Is it a duplicated question?**
~~Guess not~~ Kindof related to #518

**Is your feature request related to a problem? Please describe.**
A lot of status elements are not measurable by just "online/offline". For example we have a lot of background processes where the actual warning sign for an exited queue worker would be a queue filling up. This happens on a database level and the queue worker is not exposed by design. What we do is to have a status endpoint which exposes some values as json, for example the queue item count.

**Describe the solution you'd like**
A perfect way to enable monitoring like this would be the possibility to get a specific url, parse values (either by type, like bool/int, or using a regex) and have a threshold for a warning.

**Describe alternatives you've considered**
ELK Stack. Doable but a huge overload for a simple task. Also we could have additional status endpoints that would return 200 only if everything is fine, but this forces the monitoring level into the applications which is kind of not nice.

**Additional context**
Example monitoring values (GET request on the status enpoint, something lik https://myapp.com/appstatus.json)

```json
{
  "app_version":"fc12f05",
  "database_reachable":true,
  "laravel_version":"8.63.0",
  "queued_jobs":12,
  "failed_jobs":0,
  "php_version":"8.0.10",
  "report_generated_at":"2021-10-22 12:53:54"
}
```
We would like to get warning when `database_reachable` goes to false and when `queue_jobs` is more than 100.

**Comment by zblesk** [comment]
I would also appreciate this feature: in my case, I want to monitor a system I can make no changes to (a third-party app), but that already exposes a JSON endpoint with a number of queue items. 

(I do not need history of the value, or any other monitoring, so a grafana stack or similar would be an overkill and completely unnecessary. I only need a check in the style of 'if this json value is over X, indicate failure.")

**Comment by UmutAlihan** [comment]
Hi, this feature would be a statping killer :)

Currently I am migrating my monitors to uptime-kuma and only feature missing is this returned json based monitoring with regex.

For example I have a selenium grid deployment and would like to check a json response from /health/status endpoint, which returns a large json. I only need to see this key/value pair in the json:

"message": "Selenium Grid ready."

The below regex syntax provides this control with Statping.
(message)": "((\\"|[Selenium Grid ready.])*)"

I am currently only checking for 200 status code. But I would love to see readiness monitoring as in this example at uptime-kuma too.

Cheers,

**Comment by Firesphere** [comment]
This sounds more like a healthchecks.io problem...
https://github.com/healthchecks/healthchecks <= FOSS git repo.

Have your jobs ping success or failure, and act on that?

**Comment by Joshfindit** [comment]
I'd like to suggest having a single value in place of the ping time.

Personally I'd love JSON parsing but would be good with regex.

On the UI it would be UP/DOWN based on whether the request was successful, but then show the extracted value instead of the ping time.

I'm open to the argument that it mixes business domains, but for my usage I'm looking at "server processed X items in the past hour" with a warning if it's down and otherwise a glanceable graph showing it's working as expected.

**Comment by chakflying** [comment]
Can be considered closed with #3253.

**Comment by CommanderStorm** [comment]
@moay (ping, as only you/Louis can close issues)
Could you close this issue?
We forgot to close it during the merging of https://github.com/louislam/uptime-kuma/pull/3253 which is included in the `1.23`-release.

If you agree, could you please close this Issue, as stale issues only create immortal zombies and are really hard to issue-manage.
If not, what makes this issue different from the PR Nelson referenced? ^^

**Comment by CommanderStorm** [comment]
Resolved by https://github.com/louislam/uptime-kuma/pull/3253
